### PR TITLE
Feature/gen2 start

### PR DESCRIPTION
**Pull request by SebastianSchildt**
This starts a new code path for Gen2 protocol support ("/" insteand of "." as path sperator)

The idea is to use the chance to clean up some old code and replace code paths one by one, and later remove the old code.

This is the initial chunk for "get" queries. It includes compatibility  with Gen1 protocol, so ideally the user does not notice any difference. 

While implementing, will document which features are working, as the specification also includes quite a bit of stuff not relevant to an in-vehicle VSS server, and is a little but fuzzy on corner cases. So I hope after everything is done we have at least a clear documentation regarding our "dialect"
 

But lets first finish merging #85

### CODE REVIEW

File: include/VSSPath.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        static VSSPath fromVSS(std::string vss); //Expect Gen2 path
+        static VSSPath fromVSSGen1(std::string vssgen1); //Expect Gen1 path
+        static VSSPath fromJSON(std::string json); //Expect json path
+        static VSSPath fromVSSAuto(std::string vss); //Auto decide for Gen1 or Gen2
+
+    private:
+        std::string vsspath;
+        std::string vssgen1path;
+        std::string jsonpath;
+        bool gen1;
```
Comment (wenwenchenbosch) [inline include/VSSPath.hpp:42]: naming... and why we need here so many member variables?

**Review by wenwenchenbosch — COMMENTED** [review]
can you commit vss gen2 model for testing?

**Comment by wenwenchenbosch** [comment]
update jsoncon? clearing necessary?

File: doc/protocol.md [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+    "timestamp": 1608377709, 
+    "value": [
+        {
+            "Vehicle/Cabin/Door/Row4/Right/Window/isOpen": "---"
+        }, 
+        {
+            "Vehicle/Cabin/Door/Row4/Right/Window/Switch": "---"
+        }, 
+        {
+            "Vehicle/Cabin/Door/Row4/Right/Window/Position": "---"
```
[the remark this answers is not in the document]
Comment (SebastianSchildt) [reply doc/protocol.md:239]: Made the long parts collapsible

File: doc/protocol.md [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+    "timestamp": 1608378219, 
+    "value": [
+        {
+            "Vehicle/Drivetrain/Transmission/DriveType": "unknown"
+        }
+    ]
+}
+```
+
+The following paths would yield the same result
```
[the remark this answers is not in the document]
Comment (SebastianSchildt) [reply doc/protocol.md]: Clarified

File: include/VSSPath.hpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        
+
+        VSSPath(std::string vss, std::string vssgen1, std::string jsonpath, bool gen1origin);
+    
+    friend inline bool operator==(const VSSPath& lhs, const VSSPath& rhs) { return (lhs.vsspath == rhs.vsspath); };
+    friend inline bool operator!=(const VSSPath& lhs, const VSSPath& rhs) { return !(lhs == rhs); };
+
+};
+
+#endif
```
[the remark this answers is not in the document]
Comment (SebastianSchildt) [reply include/VSSPath.hpp]: Done

File: include/VssDatabase.hpp [hunk 5]
```diff
@@ -60,7 +66,11 @@ class VssDatabase : public IVssDatabase {
   void setSignal(const std::string &path, jsoncons::json value);
   jsoncons::json getSignal(WsChannel& channel, const std::string &path) override;
 
+  jsoncons::json getSignal2(WsChannel& channel, const VSSPath &path, bool gen1_compat) override; //Gen2 version
```
[the remark this answers is not in the document]
Comment (SebastianSchildt) [reply include/VssDatabase.hpp]: Overloaded getSignal(), marked old version as deprecated (is still needed until everything is converted to Gen2)

File: include/VSSPath.hpp [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+class VSSPath {
+    public:
+        std::string getVSSPath() const;
+        std::string getVSSGen1Path() const;
+        std::string getJSONPath() const;
+        bool isGen1Origin() const;
+        static VSSPath fromVSS(std::string vss); //Expect Gen2 path
+        static VSSPath fromVSSGen1(std::string vssgen1); //Expect Gen1 path
+        static VSSPath fromJSON(std::string json); //Expect json path
+        static VSSPath fromVSSAuto(std::string vss); //Auto decide for Gen1 or Gen2
```
[the remark this answers is not in the document]
Comment (SebastianSchildt) [reply include/VSSPath.hpp]: Now fromVSS() is automatic, and fromVSS renamed to fromVSSGen2

File: include/VSSPath.hpp (hunk 1 above)
Comment (SebastianSchildt) [reply include/VSSPath.hpp:42]: As discussed: Only run conversions once, and store all variants for performance reasons

File: doc/protocol.md [hunk 7]
```diff
@@ -143,6 +143,9 @@ A more interesting example:
     "path": "Vehicle/Cabin/Door/*/*/IsLocked", 
     "requestId": "311508011343524723835336189899358414689",
 ```
```
Comment (wenwenchenbosch) [inline doc/protocol.md:145]: 👍

**Review by wenwenchenbosch — APPROVED** [review]

**Comment by wenwenchenbosch** [comment]
> 
> 
> update jsoncon? clearing necessary?

no license changed and not major release

**Pull request merged**